src/Yaapii.Http/Parts/Uri/QueryParams.cs

```diff
@@ -45,8 +45,8 @@ public sealed partial class QueryParams : MapInputEnvelope
                     query = query.Remove(0, 1);
                 }
                 return
-                    new MapOf(
-                        query.Split(new char[] { '=', '&' })
+                    new Yaapii.Atoms.Temp.Map.MapOf(
+                        query.Split(new char[] { '=', '&' }, System.StringSplitOptions.RemoveEmptyEntries)
```

Comment:
If a query param has an empty value, removing empty entries here will cause a test like this to fail:
```
[Fact]
public void WorksWithEmptyKey()
{
    Assert.Equal(
        "",
        new QueryParams(
            "?asdf=&qwertz=123"
        ).Apply(
            new Dictionary<string, string>()
        )["query:asdf"]
    );
}
```
How about replacing this constructor with this:
```
public QueryParams(string query) : this(
    new MapOf(() =>
    {
        if (query.StartsWith("?"))
        {
            query = query.Remove(0, 1);
        }
        var result = new MapOf(new MapInputOf());
        if (query.Length > 0)
        {
            result =
                new MapOf(
                    query.Split(new char[] { '=', '&' })
                );
        }
        return result;
    })
)
{ }
```
I think this way the temporary MapOf could also be removed.

Reply:
As @Meerownymous said in the dev-yaapii channel yesterday, empty query params are an antipattern and should be removed in the corresponding snaps. Th correct solution for this case is to wok with the default return value in Demands.

Reply:
ok, just read through those messages in dev-yaapii, makes sense